src/UI/Art/SpriteArtist/SpriteMiniMapArtist.java

```diff
@@ -87,15 +102,29 @@ private static BufferedImage generateMapImage(MapInfo mapInfo)
     return image;
   }
 
-  /**
-   * Assign a set of colors to be used when generating a minimap image.
-   * @param teamColors An array of color values, one per player
-   * NOTE: If a palette is not set manually, UIUtils.getCOColors() will be used.
-   * NOTE: If too few colors are provided, black will be used as the fallback.
-   */
-  public static void setTeamPalette(Color[] teamColors)
+  private static boolean palettesDiffer(Color[] newColors)
+  {
+    if( null == lastTeamColors || null == newColors )
+      return true;
+    if( lastTeamColors.length != newColors.length )
+      return true;
+
+    boolean colorsDiffer = false;
+    for( int i = 0; i < newColors.length; ++i )
+    {
+      if( colorsDiffer(newColors[i], lastTeamColors[i]) )
+      {
+        colorsDiffer = true;
+        break;
+      }
+    }
+
+    return colorsDiffer;
+  }
+
+  private static boolean colorsDiffer(Color c1, Color c2)
```

Comment:
I could be wrong, but doesn't the `.equals()` method do this?

Reply:
So it does.